build.rs

```diff
@@ -1,212 +1,159 @@
-use quote::ToTokens;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote, TokenStreamExt};
 use std::env;
 use std::fs::File;
 use std::io::Write;
 use std::path::Path;
-
-mod types {
-    use indexmap::IndexMap;
-    use proc_macro2::TokenStream;
-    use quote::{quote, ToTokens, TokenStreamExt};
-    use serde::{Deserialize, Deserializer};
-
-    // Manual blacklist for now. See https://github.com/dtolnay/syn/issues/607#issuecomment-475905135.
-    fn has_spanned(ty: &str) -> bool {
-        match ty {
-            "DataStruct" | "DataEnum" | "DataUnion" => false,
-            "FnDecl" => false,
-            "QSelf" => false,
-            _ => true,
-        }
+use syn_codegen::{Data, Definitions, Node, Type};
+
+// Manual blacklist for now. See https://github.com/dtolnay/syn/issues/607#issuecomment-475905135.
```

Comment:
Looking at the resolution on that issue, which mentions syn-codegen, and the fact that you added syn-codegen in this PR, is there a way we get rid of this manual blacklist now?

Reply:
The resolution on the issue is that the syn.json doesn't attempt to capture *everything* that different use cases might need, so for this use case we'd continue to need this list. I can tweak the wording:

```suggestion
// Manual blacklist; see https://github.com/dtolnay/syn/issues/607#issuecomment-475905135.
```

Reply:
Ah. I was hoping that syn.json might still be extended to include this, since there is a fair chance for blacklist getting out of sync (if some node gets a `.span()` support but we silently blacklist it).

I guess if that's not the plan, then changing the wording makes sense.